clcache.py

```diff
@@ -150,7 +150,8 @@ def getManifestHash(compilerBinary, commandLine, sourceFile):
         return getFileHash(sourceFile, additionalData)
 
     @staticmethod
-    def getIncludesContentHash(listOfHeaderHashes):
+    def getIncludesContentHash(listOfIncludes):
+        listOfHeaderHashes = [getFileHash(expandBasedirPlaceholder(fileName, baseDir)) for fileName in listOfIncludes]
         return HashAlgorithm(','.join(listOfHeaderHashes).encode()).hexdigest()
```

Comment:
I was that point 1. before be294fd3dcd. But then the tests did not work anymore and thus I decided to split it in two methods again. Testing the interface at the files list level can be done as well, but it is not trivial.

Regarding 2: you want to be able to differentiate

A.h

```
# content 1
# content 2
```

B.h

```
# content 3
```

and

A.h

```
# content 1
```

B.h

```
# content 2
# content 3
```

so you need to have whatever kind of file separator.

There is room for improvements, sure. But I wanted to have this as small as possible to get to the point where you see this kind of opportunities.